modules/nlu-testing/src/views/full/TestModal.tsx

```diff
@@ -15,38 +15,81 @@ interface Props {
   onTestCreated: (test: any) => void
 }
 
-const noneIntent = {
+const TEST_ALL_CTX = '*'
+const NONE_CTX = 'none'
+const NONE_INTENT = {
   name: 'none',
   slots: []
 } as sdk.NLU.IntentDefinition
 
-export const CreateTestModal: FC<Props> = props => {
+export const TestModal: FC<Props> = props => {
   const [utterance, setUtterance] = useState<string>((props.test && props.test.utterance) || '')
-  const [expectedIntent, setTargetIntent] = useState<sdk.NLU.IntentDefinition>(noneIntent)
+  const [expectedIntent, setExpectedIntent] = useState<sdk.NLU.IntentDefinition>(NONE_INTENT)
   const [intents, setIntents] = useState<sdk.NLU.IntentDefinition[]>([])
   const [availableCtx, setAvailableCtxs] = useState([])
   const [testingCtx, setTestingCtx] = useState('*')
   const [slotConditions, setSlotConditions] = useState<_.Dictionary<string>>({})
   const [expectedCtx, setExpectedCtx] = useState<string>('')
+  const [isEditing, setIsEditing] = useState(false)
 
   useEffect(() => {
     // tslint:disable-next-line: no-floating-promises
     props.api.fetchIntents().then(intents => {
-      setIntents([...intents, noneIntent])
+      setIntents([...intents, NONE_INTENT])
       const ctxs = _.chain(intents)
         .flatMap(i => i.contexts)
         .uniq()
         .value()
-      setAvailableCtxs([...ctxs, '*', 'none'])
+      setAvailableCtxs([...ctxs, TEST_ALL_CTX, NONE_CTX])
       setExpectedCtx(ctxs[0])
     })
   }, [])
 
+  useEffect(() => {
+    const { test } = props
+    setIsEditing(!!test)
+
+    if (!test && utterance !== '') {
+      setUtterance('')
+      setTestingCtx(TEST_ALL_CTX)
+      setExpectedCtx(NONE_CTX)
+      setExpectedIntent(NONE_INTENT)
+      setSlotConditions({})
+    }
+
+    if (test) {
+      setStateFromTest(test)
+    }
+  }, [props.test])
+
+  const setStateFromTest = (test: Test) => {
+    setUtterance(test.utterance)
+    setTestingCtx(test.context)
+
+    if (test.context === TEST_ALL_CTX) {
+      const ctxCondition = test.conditions.find(([key]) => key === 'context')
```

Comment:
```suggestion
      const ctxCondition = test.conditions.find(([key]) => key === 'context')
      
```
again just a preference